docs/contributing/database-migrations/edd.mdx

```diff
@@ -24,22 +23,66 @@ For background on this decision please see the [Evolutionary Database Design RFD
 
-### Nullable
+Martin Fowler's EDD defines two types of database changes: destructive and non-destructive
+([EDD -Transition Phase: All database changes are database refactors](https://www.martinfowler.com/articles/evodb.html#TransitionPhase)).
+A destructive change is any database change that requires an accompanying code change to continue
+working as expected. A non-destructive change is the opposite: a database change that does not
+require a code change to allow the application to continue working as expected.
 
-Database tables, views and stored procedures should almost always use either nullable fields or have
-a default value. Since this will allow stored procedures to omit columns, which is a requirement
-when running both old and new code.
+### Non-destructive Database Changes
 
-### EDD Process
+An example of a non-destructive change is almost always using either nullable fields or default
+values in database tables, views, and stored procedures. We have adopted this as a standard for any
+such changes. This will allow stored procedures to omit columns, which is a requirement when running
+both old and new code.
 
-The EDD breaks up each database migration into three phases. _Start_, _Transition_ and _End_.
+### Destructive Changes
+
+In our current release process where our database changes and our code changes are coupled, even a
+new column can even be considered a destructive change if the default value of the column is a
+non-constant value that needs to be computed from elsewhere.
+
+Martin Fowler's explanation of how to elegantly handle destructive database changes in an EDD way
+breaks up such a change into three phases: _Start_, _Transition_ and _End_.
 
 ![Refactoring Stages](./stages_refactoring.jpg)
 [https://www.martinfowler.com/articles/evodb.html#TransitionPhase](https://www.martinfowler.com/articles/evodb.html#TransitionPhase)
 
-This necessitates two different database migrations. The first migration adds new content and is
-backwards compatible with the existing code. The second migration removes content and is not
-backwards compatible with that same code prior to the first migration.
+We tweak the terminology to be more easily understandable in how EDD relates to our deployment
+processes in both our environments: our always-on application in the cloud and our self-host
+deployments. We use the terms: _Initial_ Phase (instead of _Start_), _Transition_ Phase, and
+_Finalization_ Phase (instead of _End_).
+
+#### Initial Phase
+
+- Compatible with <i>previous</i> <b>and</b> <i>next</i> application code changes
+- Represents the beginning of a database change
+- Updates our database schema to support any new functionality while also maintaining old
+  functionality
+- Supports both the previous version of code and the one being upgraded to
+- Run during upgrade
+- Must execute quickly to minimize downtime.
+
+#### Transition Phase
+
+- Compatible with <i>previous</i> <b>and</b> <i>next</i> application code changes
```

Comment:
Previous and next are quite bad names for states since they change during the phases especially after a rollback where previous and next switches place.

Reply:
Renamed for better clarity